### PR TITLE
visuals/fidget: use submodule for notification group overrides

### PR DESCRIPTION
**Pull request by FrothyMarrow**
Changes `display.overrides` to use `submodule` instead of `anything` for better eval time checks.

### CODE REVIEW

**Review by NotAShelf — APPROVED** [review]
LGTM. Is this tested?

**Comment by FrothyMarrow** [comment]
> LGTM. Is this tested?

It was tested, but I was using the wrong defaults. Fixed that and added the changelog entry, should be good to go now.

File: modules/plugins/visuals/fidget/fidget.nix [hunk 1]
```diff
@@ -176,9 +176,170 @@ in {
...
           overrides = mkOption {
-            description = "Override options from the default notification config";
-            type = attrsOf (attrsOf anything);
-            default = {rust_analyzer = {name = "rust-analyzer";};};
+            description = ''
+              Overrides the default configuration for a notification group defined in `setupOpts.notification.configs`.
+
+              if any of the fields are null, the value from the default configuration is used.
+
+              if default configuration is not defined, the following defaults are used:
```
Comment (NotAShelf) [inline modules/plugins/visuals/fidget/fidget.nix:184]: new line, capitalize first letter.

File: modules/plugins/visuals/fidget/fidget.nix [hunk 2]
```diff
@@ -176,9 +176,170 @@ in {
...
+              options = {
+                name = mkOption {
+                  description = ''
+                    Name of the group, displayed in the notification window.
+                    Can be a string or a function that returns a string.
+
+                    If a function, it is invoked every render cycle with the items list,
+                    useful for rendering animations and other dynamic content.
+
+                    for more information on the function signature, see:
```
Comment (NotAShelf) [inline modules/plugins/visuals/fidget/fidget.nix:217]: new line, capitalize first letter.

include as a note:

```
::: {.note}
some note here
:::
```

**Review by NotAShelf — CHANGES_REQUESTED** [review]
a few changes required

also rebase around main please

**Review by NotAShelf — APPROVED** [review]
LGTM, thanks

**Pull request merged**